orte/mca/rml/ofi/rml_ofi_component.c

```diff
@@ -0,0 +1,1091 @@
+                                   __FILE__,__LINE__,prov);
+            }
+            providers = (opal_value_t *)providers->super.opal_list_next;
+            /*  opal_output_verbose(1,orte_rml_base_framework.framework_output,"\n %s:%d 
+             * - Moving on to next provider provders=%x",__FILE__,__LINE__,providers);  
+            */
+        }
+    } else {
+        opal_output_verbose(10,orte_rml_base_framework.framework_output,
+                            "\n query_transports() returned Error ");
+    }
+    opal_output_verbose(10,orte_rml_base_framework.framework_output,
+                        "\n End of print_transports_query() \n");
+}
+
+
+
+
+/**
+    conduit [in]: the conduit_id that triggered the progress fn
+ **/
+__opal_attribute_always_inline__ static inline int
+orte_rml_ofi_progress(ofi_transport_conduit_t* conduit)
+{
+    ssize_t ret;
+    int count=0;    /* number of messages read and processed */
+    struct fi_cq_data_entry wc = { 0 };
+    struct fi_cq_err_entry error = { 0 };
+    orte_rml_ofi_request_t *ofi_req;
+
+    opal_output_verbose(1, orte_rml_base_framework.framework_output,
+                         "%s orte_rml_ofi_progress called for conduitid %d",
+                         ORTE_NAME_PRINT(ORTE_PROC_MY_NAME),
+                         conduit->conduit_id);
+    /**
+    * Read the work completions from the CQ.
+    * From the completion's op_context, we get the associated OFI request.
+    * Call the request's callback.
+    */
+    while (true) {
+        /* Read the cq - that triggered the libevent to call this progress fn. */
+        ret = fi_cq_read(conduit->cq, (void *)&wc, 1);
+        if (0 < ret) {
+            opal_output_verbose(1, orte_rml_base_framework.framework_output,
+                         "%s cq read for conduitid %d - wc.flags = %x",
+                         ORTE_NAME_PRINT(ORTE_PROC_MY_NAME),
+                         conduit->conduit_id, wc.flags);
+            count++;
+            // check the flags to see if this is a send-completion or receive
+            if ( (wc.flags & FI_SEND) || ((wc.flags & FI_SEND) && (wc.flags & FI_TRANSMIT_COMPLETE)) )
+            {
+                opal_output_verbose(1, orte_rml_base_framework.framework_output,
+                       "%s Send completion received on conduitid %d",
+                       ORTE_NAME_PRINT(ORTE_PROC_MY_NAME),
+                       conduit->conduit_id);
+                if (NULL != wc.op_context) {
+                    /* get the context from the wc and call the message handler */
+                    ofi_req = TO_OFI_REQ(wc.op_context);
+                    assert(ofi_req);
+                    ret = orte_rml_ofi_send_callback(&wc, ofi_req);
+                    if (ORTE_SUCCESS != ret) {
+                        opal_output(orte_rml_base_framework.framework_output,
+                        "Error returned by OFI request event callback: %zd",
+                         ret);
+                         abort();
```

Comment:
If there is an unrecoverable error, we should be invoking `opal_show_help()` to print a helpful message to the user and then exiting gracefully.  I don't think calling `abort()` here (which dumps a corefile) is social -- if this happens at scale, it could result in gigabytes (or more!) of corefiles.